x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearnToRankService.java

```diff
@@ -176,15 +177,21 @@ private QueryExtractorBuilder applyParams(QueryExtractorBuilder queryExtractorBu
             return queryExtractorBuilder;
         }
 
-        Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, templateSource, Collections.emptyMap());
-        String parsedTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(params).execute();
-        // TODO: handle missing params.
-        XContentParser parser = XContentType.JSON.xContent().createParser(parserConfiguration, parsedTemplate);
-
-        return new QueryExtractorBuilder(
-            queryExtractorBuilder.featureName(),
-            QueryProvider.fromXContent(parser, false, INFERENCE_CONFIG_QUERY_BAD_FORMAT)
-        );
+        try {
+            Script script = new Script(ScriptType.INLINE, DEFAULT_TEMPLATE_LANG, templateSource, SCRIPT_OPTIONS, Collections.emptyMap());
+            String parsedTemplate = scriptService.compile(script, TemplateScript.CONTEXT).newInstance(params).execute();
+            XContentParser parser = XContentType.JSON.xContent().createParser(parserConfiguration, parsedTemplate);
+
+            return new QueryExtractorBuilder(
+                queryExtractorBuilder.featureName(),
+                QueryProvider.fromXContent(parser, false, INFERENCE_CONFIG_QUERY_BAD_FORMAT)
+            );
+        } catch (GeneralScriptException e) {
```

Comment:
We should only do a `matchnone` if the script error is specifically because of things missing. Couldn't a `GeneralScriptException` be thrown if things were generally malformed? In the malformed case, we should 100% throw and bubble up the failure.